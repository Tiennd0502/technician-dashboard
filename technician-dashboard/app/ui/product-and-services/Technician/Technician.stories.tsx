import type { Meta, StoryObj } from '@storybook/react';

import Technician from '.';

const meta: Meta<typeof Technician> = {
  title: 'Components/Technician',
  component: Technician,
  tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof Technician>;

export const Default: Story = {
  args: {},
};
