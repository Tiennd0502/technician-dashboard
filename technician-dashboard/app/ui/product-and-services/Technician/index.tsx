import { Box, Heading, Text } from '@chakra-ui/react';
import Image from 'next/image';

import { TECHNICIAN_URL } from '@/lib/constants';

interface TechnicianProps {
  src?: string;
}

const Technician = ({ src = TECHNICIAN_URL }: TechnicianProps) => (
  <Box p='7' borderWidth='1px' borderColor='border.primary' borderRadius='md' height='100%'>
    <Heading variant='headingXl'>Add Technicians</Heading>
    <Text as='p' variant='textXxs'>
      Click on the &nbsp;
      <Text as='span' variant='textXxs' color='primary.500'>
        Email&nbsp;
      </Text>
      &amp; send the magic link to technicians
    </Text>
    <Box maxW='500px' m='0 auto'>
      <Image
        src={src}
        alt='Technicians'
        style={{
          width: '100%',
          height: 'auto',
        }}
        width={326}
        height={300}
        layout='responsive'
        blurDataURL={src}
        placeholder='blur'
        priority
      />
    </Box>
  </Box>
);

export default Technician;
