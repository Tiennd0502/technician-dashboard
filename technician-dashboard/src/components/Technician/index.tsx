import { Box, Heading, Text, Image } from '@chakra-ui/react';

const Technician = () => (
  <Box p='7' borderWidth='1px' borderColor='border.primary' borderRadius='md' height='100%'>
    <Heading variant='headingXl'>Add Technicians</Heading>
    <Text as='p' variant='textXxs'>
      Click on the &nbsp;
      <Text as='span' variant='textXxs' color='primary.500'>
        Email&nbsp;
      </Text>
      &amp; send the magic link to technicians
    </Text>
    <Image src='technician.png' alt='Technicians' m='auto' />
  </Box>
);

export default Technician;
