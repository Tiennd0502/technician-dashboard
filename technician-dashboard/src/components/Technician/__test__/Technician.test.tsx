import { render } from '@testing-library/react';

import Technician from '..';

describe('Technician test cases', () => {
  test('should render correctly', () => {
    const { container } = render(<Technician />);
    expect(container).toMatchSnapshot();
  });
});
