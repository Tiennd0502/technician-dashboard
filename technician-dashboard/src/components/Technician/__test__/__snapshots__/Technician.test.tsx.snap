// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Technician test cases should render correctly 1`] = `
<div>
  <div
    class="css-neaj8j"
  >
    <h2
      class="chakra-heading css-0"
    >
      Add Technicians
    </h2>
    <p
      class="chakra-text css-0"
    >
      Click on the  
      <span
        class="chakra-text css-ss2gmn"
      >
        Email 
      </span>
      & send the magic link to technicians
    </p>
    <img
      alt="Technicians"
      class="chakra-image css-42igfv"
      src="technician.png"
    />
  </div>
</div>
`;
